gluten-core/src/main/scala/io/glutenproject/execution/BasicScanExecTransformer.scala

```diff
@@ -88,12 +89,18 @@ trait BasicScanExecTransformer extends LeafTransformSupport with BaseDataSource
   }
 
   override protected def doValidateInternal(): ValidationResult = {
+    var fields = schema.fields
+
+    this match {
+      case transformer: FileSourceScanExecTransformer =>
+        fields = appendStringFields(transformer.relation.schema, fields)
+      case transformer: HiveTableScanExecTransformer =>
+        fields = appendStringFields(transformer.getDataSchema, fields)
+      case _ =>
```

Comment:
Do we need BatchScanExec here?

Reply:
@rui-mo  I haven't encountered it here, but in theory, as long as there is a native scan of orc, it is needed. I have added in the code.